StandardLibraryEssentials/StepForwardWithSTL/LeaderboardCreation/src/leaderboard.cpp

```
@@ -0,0 +1,115 @@
+#include "leaderboard.hpp"
+
+void Leaderboard::retrieveScores() {
+    std::ifstream file(filepath);
+    if (file.is_open()) {
+        std::string line;
+        while (getline(file, line)) {
+            std::stringstream ss(line);
+            std::string name;
+            unsigned int score;
+            getline(ss, name, ',');
+            ss >> score;
+            this->scores.insert({score, name});
+        }
+        file.close();
+    } else {
+        std::cout << "Unable to open file: " << this->filepath << std::endl;
+    }
+}
+
+void Leaderboard::addScore(unsigned int score, std::string name) {
+    scores.insert({score, name});
+    if (scores.size() == 11) {
+        scores.erase(std::prev(scores.end()));
+    }
+}
+
+void Leaderboard::saveScores() {
+    std::ofstream
+            file(filepath);
+    if (file.is_open()) {
```

same note here about failing fast